lib/FusionInventory/Agent/Inventory.pm

```diff
@@ -250,7 +250,26 @@ sub addEntry {
         $entry->{SERIALNUMBER} = $entry->{SERIAL} if !$entry->{SERIALNUMBER}
     }
 
-    push @{$self->{content}{$section}}, $entry;
+    if ($params{identity} && ref $params{identity} eq 'ARRAY') {
```

Comment:
You should better handle a uniq callback, not a list of callbacks: 'identity' parameter is not even clearly identified as a list of callbacks anywhere until you carefully read the code